kubernetes/istio/istio/src/main/java/org/arquillian/cube/istio/impl/IstioClientCreator.java

```diff
@@ -16,9 +16,7 @@
 
     public void createIstioClient(@Observes final KubernetesClient client) {
 
-        KubernetesAdapter adapter = new KubernetesAdapter(client);
-        IstioClient istioClient = new IstioClient(adapter);
-
+        IstioClient istioClient = new DefaultIstioClient(client.getConfiguration());
```

Comment:
While this is perfectly fine, It's slightly better to do it via:

     client.adapt(IstioClient.class);

With this approach you decouple the code from the actual impl and also you use a clone of the http client instance instead of creating one from config.